### ISSUE TITLE
undefined reference to symbol 'pthread_setspecific@@GLIBC_2.2.5'

### ISSUE THREAD

**Issue by unoexperto** [facebook/rocksdb#6329]
I'm having troubles building rocksdb on Ubuntu 19.10 from master.

My toolchain:
* cmake version 3.13.4
* gcc (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008

I run
```
cmake --build /home/zzz/work/sideprojects/rocksdb/cmake-build-debug --target all -- -j 8
```

and everything compiles except this
```
/usr/bin/ld: third-party/gtest-1.8.1/fused-src/gtest/libgtest.a(gtest-all.cc.o): undefined reference to symbol 'pthread_setspecific@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libpthread.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/table_reader_bench.dir/build.make:88: table_reader_bench] Error 1
make[1]: *** [CMakeFiles/Makefile2:5166: CMakeFiles/table_reader_bench.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Could you please suggest what I'm missing ? I [googled](https://stackoverflow.com/questions/25617839/undefined-reference-to-symbol-pthread-key-deleteglibc-2-2-5) but it's unclear to me what part of cmake script I should modify.

**Comment by maysamyabandeh** [comment]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://www.facebook.com/groups/rocksdb.dev

**Comment by syoliver** [comment]
@maysamyabandeh This is a build issue, experienced it too and tried to fix it in #6572.

**Comment by ajkr** [comment]
Fixed by #6572.

**Comment by bhupendrasinghsolanki** [comment]
This symbol is already available in libpthread.so.0 on the system. So just add the "-lpthread" in link like and problem will resolve on CentOS7.
